src/nsls2api/api/v1/admin_api.py

```diff
@@ -50,3 +52,74 @@ async def generate_user_apikey(username: str):
     return await generate_api_key(username)
 
 
+@router.post("/admin/slack/create-proposal-channel/{proposal_id}")
+async def create_slack_channel(proposal_id: str) -> SlackChannelCreationResponseModel:
+    proposal = await proposal_service.proposal_by_id(int(proposal_id))
+
+    if proposal is None:
+        return fastapi.responses.JSONResponse(
+            {"error": f"Proposal {proposal_id} not found"}, status_code=404
+        )
+
+    channel_name = proposal_service.slack_channel_name_for_proposal(proposal_id)
+
+    if channel_name is None:
+        return fastapi.responses.JSONResponse(
+            {"error": f"Slack channel name cannot be found for proposal {proposal_id}"},
+            status_code=404,
+        )
+
+    channel_id = await slack_service.create_channel(
+        channel_name, True, description=f"Discussion related to proposal {proposal_id}"
+    )
+
+    if channel_id is None:
+        return fastapi.responses.JSONResponse(
+            {"error": f"Slack channel creation failed for proposal {proposal_id}"},
+            status_code=500,
+        )
+
+    logger.info(f"Created slack channel '{channel_name}' for proposal {proposal_id}.")
+
+    # Store the created slack channel ID
+    proposal.slack_channel_id = channel_id
+    await proposal.save()
+
+    # Add the beamline slack channel managers to the channel
+    slack_managers_added = []
+    for beamline in proposal.instruments:
+        slack_managers = await beamline_service.slack_channel_managers(beamline)
+        logger.info(f"Adding Slack channel managers for {beamline} beamline [{slack_managers}].")
+        if len(slack_managers) > 0:
+            slack_service.add_users_to_channel(
+                channel_id=channel_id, user_ids=slack_managers
+            )
+            slack_managers_added.append(slack_managers)
+
+    # Add the users on the proposal to the channel
```

Comment:
Ahhh, this is perhaps why `create_channel()` is allowed to succeed when the channel already exists...so that the channel could be updated with the latest list of users on the proposal?

Consider making that a separate API endpoint to update the users on the channel.  In that case, it might be better to remove all non-bot users from the channel and then re-add them, because presumably users could be removed from PASS as well as added.

Reply:
Indeed - good idea on the "update channel" endpoint ... but in reality it is probably going to be almost all the code as for the create endpoint - assuming that if someone wants to update a channel and it doesn't exist - do we want to create it ? 

Maybe I should just embrace the HTTP methods more and drop the "create/update/etc" name and say POST == create and PUT == update ?

Reply:
I favor switching to HTTP verbs.  The common code could be refactored into a function that both endpoints call.

Reply:
I've created #81 to address this - I think if it's ok I will do this change as a separate PR?